Client/Redux/GlobalState/ActiveScreenState.swift

```diff
@@ -34,7 +36,17 @@ struct ActiveScreensState: Equatable {
             case .showScreen(.themeSettings):
                 screens += [.themeSettings(ThemeSettingsState())]
             case .closeScreen(.themeSettings):
-                screens = []
```

Comment:
@OrlaM @yoanarios This previous code looked incorrect but LMK if this change is wrong; I updated it from clearing all `screens` to only removing screens of the specific type.

Reply:
Yes the idea is to filter out the screen that it was closed but for the sake of the example I just set the array to empty and forgot to update de code afterwards. We could do something more generic without checking the screen that it was closed

Reply:
> We could do something more generic without checking the screen that it was closed

The difficulty is that it's comparing two separate enums (even though the cases have the same name). The `.closeScreen()` associated value is of type `AppScreen`, while the `screens` collection itself contains instances of `AppScreenState`. So we define the screens in two places (`AppScreenState` and `AppScreen`), but there's no automatic way for Swift to know the relationship between those (afaik).

We could add a helper function to `AppScreenState` to return the matching enum value of `AppScreen` and then we could use that to filter the screens. Something like this (below).

LMK if that seems reasonable or if you have other ideas. 

```swift
enum AppScreenState: Equatable {
    case themeSettings(ThemeSettingsState)
    case remoteTabsPanel(RemoteTabsPanelState)

    static let reducer: Reducer<Self> = { state, action in
        …
    }

    var associatedAppScreen: AppScreen {
        switch self {
        case .themeSettings: return .themeSettings
        case .remoteTabsPanel: return .remoteTabsPanel
        }
    }
}
```

(And then to handle the `closeScreen` action we could just do):

```swift
if let action = action as? ActiveScreensStateAction {
            switch action {
            case .closeScreen(let screenType):
                screens = screens.filter({ return $0.associatedAppScreen != screenType })
…
```

Reply:
Sounds good to me 👍